src/analysis/normed/group/seminorm.lean

```diff
@@ -136,13 +183,32 @@ end
 
 end group_seminorm_class
 
+section nonarch_add_group_seminorm_class
+variables [add_group E] [nonarch_add_group_seminorm_class F E] (f : F) (x y : E)
+include E
+
+lemma map_sub_le_max : f (x - y) ≤ max (f x) (f y) :=
+by { rw [sub_eq_add_neg, ← map_neg_eq_map' f y], exact map_add_le_max _ _ _ }
+
+end nonarch_add_group_seminorm_class
+
 @[to_additive, priority 100] -- See note [lower instance priority]
 instance group_seminorm_class.to_nonneg_hom_class [group E] [group_seminorm_class F E] :
   nonneg_hom_class F E ℝ :=
 { map_nonneg := λ f a, nonneg_of_mul_nonneg_right
     (by { rw [two_mul, ←map_one_eq_zero f, ←div_self' a], exact map_div_le_add _ _ _ }) two_pos,
   ..‹group_seminorm_class F E› }
 
+@[priority 100] -- See note [lower instance priority]
+instance nonarch_group_seminorm_class.to_nonneg_hom_class [add_group E]
+  [nonarch_add_group_seminorm_class F E] :
+  nonneg_hom_class F E ℝ :=
+{ map_nonneg := λ f a, begin
+    rw [← map_zero f, ← sub_self a],
+    exact le_trans (map_sub_le_max _ _ _) (by rw max_self (f a)),
+  end,
+  ..‹nonarch_add_group_seminorm_class F E› }
+
```

Comment:
```suggestion
```

Reply:
I am using this and `nonarch_add_group_seminorm_class.to_zero_hom_class` in the proof of `nonarch_add_group_seminorm_class.to_add_group_seminorm_class`. It seems to me that if I don't do these first, I would have to re-prove some lemmas inside that proof.

Reply:
I think you could just do `have map_nonneg : ∀ a, 0 ≤ f a` within the proof.